server/src/assets/game/tableGenerator/converter.py

```diff
@@ -1,127 +1,184 @@
-            stopSegment = None
-            segments = []
-            isError = False
-
-            for qblock in configuratonBahn["platforms"]:
-                if qblock["id"] == lastBlock:
-                    for segment in qblock["main"]:
-                        segments.append(segment)
-                    break
-            for qblock in configuratonBahn["blocks"]:
-                if qblock["id"] == lastBlock:
-                    for segment in qblock["main"]:
-                        segments.append(segment)
-                    break
-            if len(segments) == 1:
-#                print("Segment: {} at Block {}".format(segments[0], lastBlock))
-                stopSegment = segments[0]
-            elif len(segments) == 2:
-                if segments[0][0:-1] == segments[1][0:-1]:
- #                   print(segments)
-                    stopSegment = segments[0][0:-1]
-#                    print(stopSegment)
-                else:
-                    print("Error Segment {} and Segment {} are not the same. Block {}".format(segments[0], segments[1], block["id"]))
-                    isError = True
-            else:
-                print("Error with BlockID {}".format(block["id"]))
-                isError = True
-            if isError:
-                break
-            resultData[block["id"]][destination] = {}
-            resultData[block["id"]][destination]["route-id"] = routeID
-            resultData[block["id"]][destination]["orientation"] = orientation
-            resultData[block["id"]][destination]["block"] = lastBlock
-            resultData[block["id"]][destination]["segment"] = stopSegment
-
-# Sort
-originalResultData = resultData
-resultData = {}
-for block in originalResultData:
-    destinations = []
-    for destination in originalResultData[block]:
-        destinations.append(destination)
-    destinationsSorted = []
-
-    with open(groupingFile, "r") as csvFile:
+import json, yaml, os
+
+##### GLOBALS #####
+pathToConfig = "../../../../../configurations"
+
+interlockingTableFileName = "interlocking_table.yml"
+extrasConfigFileName = "extras_config.yml"
+
+blacklistFileDirectory = "./blacklists"
+signalFlagCSVFileDirectory = "./flagMappings"
+
+##### FUNCTIONS #####
+def getConfigurationsWithMatchingSignalFlagCSV() -> list[str]:
+    """
+    For every configuration folder in swtbahn-cli/configurations, check if a corresponding
+    signal-to-symbol/flag mapping CSV file exists. Returns a list with names of all folders
+    where this CSV file exists.
+    """
+    global pathToConfig, signalFlagCSVFileDirectory
+    # Load the list of available CSV files with signal-symbol/flag map definitions
+    signalFlagCSVFolderItemList = os.scandir(signalFlagCSVFileDirectory)
+    flagMappingFileNames = []
+    for entry in signalFlagCSVFolderItemList:
+        if entry.is_file() and entry.name[-4:] == ".csv":
+            flagMappingFileNames.append(entry.name[:-4])
```

Comment:
Could rewrite as a list comprehension:
```python
flagMappingFileNames = [entry.name[:-4] for entry in signalFlagCSVFolderItemList if entry.is_file() and entry.name[-4:] == ".csv"]
```

Reply:
Will implement the suggestion. Performance is not a huge concern here though, we run this script maybe once a semester, and loading the yaml content elsewhere in this script takes a few seconds already, this part is comparatively quick.